File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/jdbc/AbstractJdbcRepositoryTest.java
package ru.tinkoff.kora.database.common.annotation.processor.jdbc;

import org.intellij.lang.annotations.Language;
import ru.tinkoff.kora.database.common.annotation.processor.AbstractRepositoryTest;

import java.util.List;

public abstract class AbstractJdbcRepositoryTest extends AbstractRepositoryTest {
    protected MockJdbcExecutor executor = new MockJdbcExecutor();

    @Override
    protected String commonImports() {
        return super.commonImports() + """
            import ru.tinkoff.kora.database.common.annotation.*;
            import ru.tinkoff.kora.database.common.*;
            import ru.tinkoff.kora.database.jdbc.*;
            import ru.tinkoff.kora.database.jdbc.mapper.result.*;
            import ru.tinkoff.kora.database.jdbc.mapper.parameter.*;
            import ru.tinkoff.kora.common.Mapping;

            import java.sql.*;
            import javax.annotation.Nullable;
            """;
    }

    protected TestObject compileJdbc(List<?> arguments, @Language("java") String... sources) {
        return this.compile(this.executor, arguments, sources);
    }
}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/jdbc/AbstractJdbcRepositoryTest.kt
package ru.tinkoff.kora.database.symbol.processor.jdbc

import org.intellij.lang.annotations.Language
import ru.tinkoff.kora.database.symbol.processor.AbstractRepositoryTest

abstract class AbstractJdbcRepositoryTest : AbstractRepositoryTest() {
    val executor = MockJdbcExecutor()

    override fun commonImports(): String {
        return super.commonImports() + """
            import ru.tinkoff.kora.database.jdbc.*;
            import ru.tinkoff.kora.database.jdbc.mapper.result.*;
            import ru.tinkoff.kora.database.jdbc.mapper.parameter.*;
            import ru.tinkoff.kora.common.Mapping;

            import java.sql.*;
        """.trimIndent()
    }

    protected fun compile(arguments: List<*>, @Language("kotlin") vararg sources: String) = compile(executor, arguments, *sources)
}
